write-fonts/src/tables/layout.rs

```diff
@@ -229,6 +229,27 @@ impl ClassDef {
         one.into_iter().flatten().chain(two.into_iter().flatten())
     }
 
+    /// Return the glyph class for the provided glyph.
+    ///
+    /// Glyphs which have not been assigned a class are given class 0
+    pub fn get(&self, glyph: GlyphId) -> u16 {
+        self.get_raw(glyph).unwrap_or(0)
+    }
+
+    // used for testing
+    fn get_raw(&self, glyph: GlyphId) -> Option<u16> {
+        match self {
+            ClassDef::Format1(table) => glyph
+                .to_u16()
+                .checked_sub(table.start_glyph_id.to_u16())
+                .and_then(|idx| table.class_value_array.get(idx as usize))
+                .copied(),
+            ClassDef::Format2(table) => table.class_range_records.iter().find_map(|rec| {
+                (rec.start_glyph_id <= glyph && rec.end_glyph_id <= glyph).then_some(rec.class)
+            }),
+        }
+    }
+
     pub fn class_count(&self) -> u16 {
         //TODO: implement a good integer set!!
```

Comment:
This wants to be an issue, presumably one that says "make me a memory safe hb_set"?

Reply:
#192